batch/batch/cloud/gcp/resource_utils.py

```diff
@@ -1,33 +1,247 @@
 import logging
 import math
-import re
 from typing import Optional, Tuple
 
 log = logging.getLogger('utils')
 
 GCP_MAX_PERSISTENT_SSD_SIZE_GIB = 64 * 1024
-MACHINE_TYPE_REGEX = re.compile('(?P<machine_family>[^-]+)-(?P<machine_type>[^-]+)-(?P<cores>\\d+)')
-GCP_MACHINE_FAMILY = 'n1'
-
-MACHINE_FAMILY_TO_ACCELERATOR_VERSIONS = {'g2': 'l4'}
-
-MACHINE_FAMILY_TO_NUM_GPUS = {
-    'g2-standard-4': 1,
-    'g2-standard-8': 1,
-    'g2-standard-12': 1,
-    'g2-standard-16': 1,
-    'g2-standard-32': 1,
-    'g2-standard-24': 2,
-    'g2-standard-48': 4,
-    'g2-standard-96': 8,
-}
 
+GCP_MACHINE_FAMILY = 'n1'
 
 MEMORY_PER_CORE_MIB = {
     ('n1', 'standard'): 3840,
     ('n1', 'highmem'): 6656,
     ('n1', 'highcpu'): 924,
-    ('g2', 'standard'): 4000,
+}
+
+
+class GPUConfig:
+    def __init__(self, num_gpus: int, gpu_type: str):
+        self.num_gpus = num_gpus
+        self.gpu_type = gpu_type
+
+
+class MachineTypeParts:
+    def __init__(self, machine_family: str, worker_type: str, cores: int, memory: int, gpu_config: Optional[GPUConfig]):
+        self.machine_family = machine_family
+        self.worker_type = worker_type
+        self.cores = cores
+        self.memory = memory
+        self.gpu_config = gpu_config
+
+
+# highcpu n1 machines are rounded up to 924 mib per core
+MACHINE_TYPE_TO_PARTS = {
+    'n1-standard-1': MachineTypeParts(
+        cores=1,
+        memory=int(3.75 * 1024**3),
```

Comment:
nit: you could make a `gib` function that just does `int(x * 1024**3)`. would make this a little more readable and less work to make sure all of these arithmetics are correct